Module1.TypesAndClasses/Module1.TypesAndClasses.Tests/ShapesTests.cs

```diff
@@ -55,58 +59,63 @@ public void TestShapesToString()
         {
             var shapes = new List<IShape>
             {
-                new Circle(1),
                 new Ellipse(2,2),
                 new EquilateralTriangle(5),
-                new Rectangle(4,4)
+                new Rectangle(4,4, Units.meters),
+                new Rectangle(40,50, Units.centimeters),
+                new Rectangle(60,40, Units.centimeters),
+                new Rectangle(40,20, Units.centimeters),
+                new Rectangle(4000,8000, Units.millimeters)
             };
-
-
+           
             foreach (var shape in shapes)
             {
                 Assert.Equal($"Shape: '{shape.GetType().Name}'. Square = {shape.Square()}, perimeter = {shape.Perimeter()}", shape.ToString());
             }

```

Comment:
asserting inside a controller is not a good practice, please create a separate method for asserts if they are needed